packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -50,7 +50,7 @@ typedef FrameCallback = void Function(Duration timeStamp);
 ///
 /// The type argument `T` is the task's return value. Consider `void` if the
 /// task does not return a value.
-typedef TaskCallback<T> = T Function();
+typedef TaskCallback<T> = FutureOr<T> Function();
```

Comment:
We should add some documentation in scheduleTask method to mention the behavior.

Reply:
Thanks! Looking at `scheduleTask`, the doc comment mentions:

> Schedules the given task with the given priority and returns a [Future](https://api.flutter.dev/flutter/dart-async/Future-class.html) that completes to the task's eventual return value.

This seems to already imply that the task will complete (irregardless of whether the task is async or not). In that case do you think we should still need to update it?

Reply:
`eventual return value` part is a bit vague. maybe add phrase to describe the behavior more clearly.

Reply:
Done, would you like to take a look again?